### PR TITLE
refactor(datasource/nuget): move v2/v3 API logic to classes

### PR DESCRIPTION
**Pull request by fgreinacher**
## Changes

This is a preparation step for being able to use cache decorators more easily

## Context

See https://github.com/renovatebot/renovate/pull/28071#discussion_r1536813935

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/modules/datasource/nuget/common.ts [hunk 1]
```diff
@@ -47,3 +48,23 @@ export function parseRegistryUrl(registryUrl: string): ParsedRegistryUrl {
...
   return { feedUrl, protocolVersion };
 }
+
+/**
+ * Compare two versions. Return:
+ * - `1` if `a > b` or `b` is invalid
+ * - `-1` if `a < b` or `a` is invalid
+ * - `0` if `a == b` or both `a` and `b` are invalid
+ */
+export function sortNugetVersions(a: string, b: string): number {
```
Comment (fgreinacher) [inline lib/modules/datasource/nuget/common.ts:58]: Not really v2/v3-specific, so I move it here.

File: lib/modules/datasource/nuget/v2.ts [hunk 2]
```diff
@@ -6,65 +6,71 @@ import { regEx } from '../../../util/regex';
...
+    const dep: ReleaseResult = {
+      releases: [],
+    };
+    let pkgUrlList: string | null = `${feedUrl.replace(
+      regEx(/\/+$/),
+      '',
+    )}/FindPackagesById()?id=%27${pkgName}%27&$select=Version,IsLatestVersion,ProjectUrl,Published`;
+    while (pkgUrlList !== null) {
+      // typescript issue
+      const pkgVersionsListRaw: HttpResponse<string> =
```
Comment (viceice) [inline lib/modules/datasource/nuget/v2.ts:28]: do we still need the explicit type here?

**Review by viceice — COMMENTED** [review]
otherwise LGTM

File: lib/modules/datasource/nuget/v2.ts (hunk 2 above)
Comment (fgreinacher) [reply lib/modules/datasource/nuget/v2.ts]: Sharp eyes! Nope, that is not required. Removed in b152c05d02e365723cd83719928cc99971bc7fd2

**Review by viceice — APPROVED** [review]

**Pull request merged**